google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py

```diff
@@ -48,14 +48,14 @@
 @listens_for(Pool, "reset")
 def reset_connection(dbapi_conn, connection_record):
     """An event of returning a connection back to a pool."""
-    if dbapi_conn.connection.inside_transaction:
-        dbapi_conn.connection.rollback()
+    if isinstance(dbapi_conn.connection, spanner_dbapi.Connection):
```

Comment:
Resetting attributes only in case it's a Spanner connection. Otherwise doing `rollback()`, which is the default behavior of the resetting method.